add initial travis file

hornc opened the pull request:
@mekarpeles I don't have access to enable travis-ci but it's a good idea raised in #17 

Just getting the ball rolling with this simple starter yml . It'll most likely need the other dependencies installed before everything works nicely.

tfmorris commented:
I've actually have a travis file on my fork as well. I hadn't finished debugging the Mac build which is why there's no PR for it.

You can see the builds here: https://travis-ci.org/tfmorris/openlibrary-client

hornc commented:
Ah, thanks @tfmorris. I didn't know osx was available for Travis. From a quick dig around it looks like there are issues with python and osx, and seemingly this issue from 2014 is still current and the official status?: https://github.com/travis-ci/travis-ci/issues/2312 . "Python runtime is not yet available on OS X"

tfmorris commented:
Yes, I think "unsupported" is still the official status, although I'm
hopeful that the recipe here will work:
https://github.com/travis-ci/travis-ci/issues/2312#issuecomment-289140791

mekarpeles commented:
@hornc @tfmorris should I hold off on review until we have something for the mac build? Or should I accept and then we can play the changes for mac over it?

mekarpeles commented:
p.s. thank you both for your contributions!

tfmorris commented:
You shouldn't wait for Mac, but I haven't compared the two Travis files to
see if they differ in other ways.

hornc commented:
@mekarpeles and @tfmorris , I have updated this PR now that all @tfmorris's test fixes are merged, and added a custom repo to get yaz 5.22 for Ubuntu so that those marc tests pass, and CI uses the same version as OSX.

We may need to add something in the readme to say yaz 5 is required.  I couldn't find a nice clear how-to , but http://ftp.indexdata.dk/pub/yaz/ubuntu/README is what I used to get this working.

I have not tried to get osx working in this PR. The recipe @tfmorris looks like it could do it though, but let's add that in another PR. I _think_ the environments are comparable now that the same version of yaz is installed.

I'm using all the pythons from @tfmorris's travis branch, and they are all passing now: https://travis-ci.org/hornc/openlibrary-client/builds/256264950

@mekarpeles you'll need to enable travis-ci on the repo for the IA organization.

mekarpeles commented:
Enabled Travis-ci integration and merging

tfmorris commented:
I'm seeing "Not an active repository" when I go to
https://travis-ci.org/internetarchive/openlibrary-client/

mekarpeles commented:
Thanks, you're right -- I activated it from Tavis.ci and it looks to me like the last build passed.
